src/cpp/session/resources/schema/user-prefs-schema.json

```diff
@@ -943,6 +943,42 @@
             "title": "Show hidden files in Files pane",
             "description": "Whether to show hidden files in the Files pane."
         },
+        "always_shown_files": {
+            "type": "array",
+            "description": "List of files that are always shown in the Files Pane, regardless of whether hidden files are shown",
+            "title": "Files always shown in the Files Pane",
+            "items": {
+                "type": "string"
+            },
+            "default": [
+                ".build.yml",
+                ".gitlab-ci.yml",
+                ".travis.yml"
+            ]
+        },
+        "always_shown_extensions": {
```

Comment:
If the user tries to set this preference, will it override the set of default always-show extensions? What's the right way for a user to add to the set of RStudio defaults?

In addition, if the user starts setting their own values here, does that imply that if we change this list in the future they won't see the updated default? I could see this being potentially problematic if a new dotfile became important in projects, but it wasn't visible to users who had already modified this preference.

Reply:
I don't see this as a user-facing feature, as there's no UI you'd have to be _very_ motivated to use it.

This list will wind up in the Admin Guide via our auto docs builder, so admins can start with that if they want to purely extend the list instead of replace it. It does mean that you're effectively on your own once you've customized the option. It'd be cool if there were some way to e.g., say "start with RStudio's built in list and add these 3", but empirically I've never seen anyone ask for that.